tests/testthat/test-sampler-random-walk.R

```diff
@@ -26,18 +17,9 @@ test_that("validate construction of rw sampler", {
 
 
 test_that("validate sampler against model on initialisation", {
-  m <- local({
-    shape <- 1
-    rate <- 1
-    mcstate_model(
-      parameters = "gamma",
-      direct_sample = function() rgamma(1, shape = shape, rate = rate),
-      density = function(x) dgamma(x, shape = shape, rate = rate, log = TRUE),
-      gradient = function(x) (shape - 1) / x - shape,
-      domain = rbind(c(0, Inf)))
-  })
+  m <- ex_simple_gamma1()
```

Comment:
Makes no difference to test passing - 
but should this be `model`, not `m` for lines 26/28?

Reply:
yup, fixed for clarity.  the reason why it never works is because we never force the promise (we fail before `model` was ever referenced, as in:

```
f <- function(x) 1
f(whatever)
```